database/seeders/DefaultSkills.php

```diff
@@ -29,7 +86,42 @@ public function run(): void
             ['skill_name' => 'Electronics safety', 'category' => 2],
             ['skill_name' => 'Replacing screens', 'category' => 2],
         ];
+    }
+
+    /**
+     * Seed skills to database based on configuration
+     */
+    private function seedSkills(array $data): void
+    {
+        $truncate = env('SEEDING_TRUNCATE_SKILLS', true);
+
+        if ($truncate) {
+            Log::info('Truncating and re-inserting all skills');
+            DB::table('skills')->truncate();
+            DB::table('skills')->insert($data);
+            Log::info('Inserted ' . count($data) . ' skills');
+            return;
+        }
+
+        Log::info('Merging new skills only');
+        $added = 0;
+        $modified = 0;
+
+        foreach ($data as $entry) {
+            $skill = DB::table('skills')->where('skill_name', $entry['skill_name'])->first();
```

Comment:
Trying to get my brain wrapped around laravel db objects. Since there's a `Skill` Model, can't this be:

```
$skill = Skill::where('skill_name', $entry['skill_name'])->first();
```

or maybe:

```
$skill = Skill::firstWhere('skill_name', $entry['skill_name']);
```

Reply:
It also looks like there is a `firstOrCreate` method that may be handy. I don't know about here because it will be hard to keep track of the `$added` value.

Reply:
The `$skill` variable is just an array. I thought about using the `Skills` model which is an [Eloquent ORM](https://laravel.com/docs/12.x/eloquent), but thought its not worth it here. This is a script that would get ran as part of the init process and not really part of the core app, so it didn't make since to integrate the `App/Models` namespace here.

Reply:
How is the `$skill` var an array if you are accessing properties? i.e. `$skill->description`?

In php, you can't access array values with the arrow notation.

<img width="770" alt="image" src="https://github.com/user-attachments/assets/cd961b03-b2be-47fa-8bf4-ab80ffad6106" />

Reply:
Ah my bad! That's correct, it's an `stdClass` object not an array.

That said, the thought behind not using the `Skills` model still stand.